tfx_bsl/beam/run_inference_arrow.py

```diff
@@ -117,17 +116,25 @@ def RunInferenceImpl(  # pylint: disable=invalid-name
   """
   logging.info('RunInference on model: %s', inference_spec_type)
 
-  batched_examples = examples | 'BatchExamples' >> beam.BatchElements()
+  # TODO (Maxine): uncomment this once we change the api to take input 
+  # Union[tf.train.Example, tf.train.SequenceExample]
+  # data_type = _get_data_type(examples)
+
+  data_type = DataType.EXAMPLE
   operation_type = _get_operation_type(inference_spec_type)
   if operation_type == OperationType.CLASSIFICATION:
```

Comment:
Could we have a separate PTransform to do the following things:
1) Detect data type and error out for classify, regress and multihead if data type is not tf.Example?
2) Translate RecordBatch to serialized example
In this case, all the _Batch*DoFn and _Remote*DoFn could use Text as input instead of RecordBatch.

At the same time, we can consider how to merge this with current RunInferenceImpl implementation.